api/src/main/resources/liquibase.xml

```diff
@@ -328,4 +328,25 @@
             TIMESTAMP(CURDATE()), 'MM/dd/yyyy HH:mm:ss', 86400, TRUE, 1, NOW(), UUID());
         </sql>
     </changeSet>
+
+    <changeSet id="queue_entry_20230809-01" author="cynthiakamau">
+        <preConditions onError="WARN" onFail="MARK_RAN">
+            <tableExists tableName="queue_entry"/>
+            <not><columnExists tableName="queue_entry" columnName="location_coming_from"/></not>
+        </preConditions>
+        <comment>
+            Add column location_coming_from to queue entry table
+        </comment>
+        <addColumn tableName="queue_entry">
+            <column name="location_coming_from" type="int">
+            </column>
+        </addColumn>
+    </changeSet>
+
+    <changeSet id="queue_entry_20230809-02" author="cynthiakamau">
+        <preConditions onFail="MARK_RAN">
+            <columnExists tableName="queue_entry" columnName="location_coming_from"/>
+        </preConditions>
+        <addForeignKeyConstraint baseColumnNames="queue_entry" baseTableName="location_coming_from" constraintName="queue_entry_location_coming_from_id_fk" deferrable="false" initiallyDeferred="false" onDelete="NO ACTION" onUpdate="NO ACTION" referencedColumnNames="queue_id" referencedTableName="queue"/>
```

Comment:
In the addForeignKeyConstraint statement, there are a couple of issues I see:
- The baseColumnNames should be `location_coming_from` not `queue_entry`. This specifies the column in the queue_entry table that the foreign key refers to.
- The baseTableName should be `queue_entry` not `location_coming_from`. This specifies the table that contains the column the foreign key constraint is on.

Reply:
Both `deferrable="false"` and `initiallyDeferred="false"` are redundant since default values are false

Reply:
Makes sense. Sorry I didn't pay much attention